src/pages/workspaces/workspace/_tests/WorkspaceMenu.test.js

```diff
@@ -0,0 +1,258 @@
+import '@testing-library/jest-dom'
+
+import { fireEvent, render, screen } from '@testing-library/react'
+import { useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { MenuTrigger } from 'src/components/PopupTrigger'
+import TooltipTrigger from 'src/components/TooltipTrigger'
+import { useWorkspaceDetails } from 'src/components/workspace-utils'
+import WorkspaceMenu from 'src/pages/workspaces/workspace/WorkspaceMenu'
+
+
+jest.mock('src/components/workspace-utils', () => {
+  const originalModule = jest.requireActual('src/components/workspace-utils')
+  return {
+    ...originalModule,
+    useWorkspaceDetails: jest.fn()
+  }
+})
+
+jest.mock('src/components/PopupTrigger', () => {
+  const originalModule = jest.requireActual('src/components/PopupTrigger')
+  return {
+    ...originalModule,
+    MenuTrigger: jest.fn()
+  }
+})
+
+jest.mock('src/components/TooltipTrigger', () => ({
+  ...jest.requireActual('src/components/TooltipTrigger'),
+  __esModule: true,
+  default: jest.fn()
+}))
+
+
+describe('test workspace undefined', () => {
+  beforeEach(() => {
+    useWorkspaceDetails.mockReturnValue({ workspace: undefined })
+    MenuTrigger.mockImplementation(({ content }) => { return div([content]) })
+    TooltipTrigger.mockImplementation(({ content, children }) => {
+      const [open, setOpen] = useState(false)
+      return (div([
+        div(
+          {
+            onMouseEnter: () => {
+              setOpen(true)
+            },
+            onMouseLeave: () => {
+              setOpen(false)
+            }
+          },
+          [children]
+        ),
+        open && !!content && div([content])
+      ]))
+    })
+  })
+
+  it('tests disabled state for clone menu item when workspace is undefined', () => {
```

Comment:
it() labels refer to the thing being tested (workspace menu with undefined workspace here). So "renders Clone menu item as disabled" would work here.